core/scripts/umip-3/4_ExecuteSimulate.js

```diff
@@ -0,0 +1,481 @@
+
+  const identifier = pendingRequests[0].identifier.toString();
+  const time = pendingRequests[0].time.toString();
+  const price = "1"; // a "yes" vote to include the new EMP factory
+
+  const salt = getRandomUnsignedInt();
+
+  // We can now use the new create vote hash function
+  const voteHash = computeVoteHash({
+    price,
+    salt,
+    account: foundationWallet,
+    time: time,
+    roundId: currentRoundId,
+    identifier
+  });
+
+  /** *******************************************************************************
+   * 6) Submit a commitment to register the emp factory and identifer with the DVM  *
+   **********************************************************************************/
+  console.log("6. COMMIT VOTE FROM FOUNDATION WALLET\nVote information:");
+  console.table({
+    price: price,
+    salt: salt.toString(),
+    account: foundationWallet,
+    time: time,
+    roundId: currentRoundId,
+    identifier: identifier,
+    voteHash: voteHash
+  });
+  // send the foundation wallet some eth to submit the Tx
+  await web3.eth.sendTransaction({ from: accounts[0], to: foundationWallet, value: web3.utils.toWei("1") });
+
+  const VoteTx = await voting.commitVote(identifier, time, voteHash, { from: foundationWallet });
+  console.log("Voting Tx done!", VoteTx.tx);
+
+  /** *****************************
+   * 7) Revealing foundation vote *
+   ********************************/
+  console.log("7. REVEALING FOUNDATION VOTE");
+  currentTime = (await voting.getCurrentTime()).toNumber();
+  await advanceBlockAndSetTime(web3, currentTime + secondsPerDay);
+
+  console.log(
+    "⏱  Advancing time to move to next voting round to enable reveal\nNew timestamp:",
+    currentTime,
+    "voting phase",
+    (await voting.getVotePhase()).toNumber(),
+    "currentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+
+  const revealTx = await voting.revealVote(identifier, time, price, salt, { from: foundationWallet });
+  console.log("Reveal Tx done!", revealTx.tx);
+
+  currentTime = (await voting.getCurrentTime()).toNumber();
+  await advanceBlockAndSetTime(web3, currentTime + secondsPerDay);
+
+  console.log(
+    "⏱  Advancing time to move to next voting round to conclude vote\nNew timestamp:",
+    currentTime,
+    "voting phase",
+    (await voting.getVotePhase()).toNumber(),
+    "currentRoundId",
+    (await voting.getCurrentRoundId()).toString()
+  );
+
+  assert.equal((await voting.getPendingRequests()).length, 0); // There should be no pending requests as vote is concluded
+
+  /** ******************************************************************
+   * 8) Execute proposal submitted to governor now that voting is done *
+   *********************************************************************/
+  console.log("8. EXECUTING GOVERNOR PROPOSALS");
+  console.log((await governor.numProposals()).toNumber());
+  const proposalId = 0; // first and only proposal in voting.sol
```

Comment:
Just confirm that this is the correct proposal index.

Reply:
yup. it is as this script assumes that the upgrade has deployed a new DVM.